tao_compiler/mlir/disc/transforms/mhlo_rewriters.cc

```diff
@@ -0,0 +1,89 @@
+// Copyright 2021 The BladeDISC Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+#include "mlir-hlo/Dialect/mhlo/IR/hlo_ops.h"
+#include "mlir-hlo/utils/hlo_utils.h"
+#include "mlir/Dialect/Arithmetic/IR/Arithmetic.h"       // from @llvm-project
+#include "mlir/Dialect/Func/IR/FuncOps.h"                // from @llvm-project
+#include "mlir/Dialect/Tensor/IR/Tensor.h"               // from @llvm-project
+#include "mlir/IR/Attributes.h"                          // from @llvm-project
+#include "mlir/IR/Location.h"                            // from @llvm-project
+#include "mlir/IR/MLIRContext.h"                         // from @llvm-project
+#include "mlir/IR/Operation.h"                           // from @llvm-project
+#include "mlir/IR/PatternMatch.h"                        // from @llvm-project
+#include "mlir/Pass/Pass.h"                              // from @llvm-project
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"  // from @llvm-project
+#include "mlir/Transforms/Passes.h"                      // from @llvm-project
+#include "stablehlo/dialect/ChloOps.h"
+#include "transforms/PassDetail.h"
+
+namespace mlir {
+namespace disc_ral {
+
+namespace {
+struct BatchNormInferenceOpConvert
+    : public OpRewritePattern<mhlo::BatchNormInferenceOp> {
+  explicit BatchNormInferenceOpConvert(MLIRContext* context)
+      : OpRewritePattern(context) {}
+
+  LogicalResult matchAndRewrite(mhlo::BatchNormInferenceOp op,
+                                PatternRewriter& rewriter) const override {
+    Value eps = chlo::getConstantLike(rewriter, op.getLoc(), op.epsilon(),
```

Comment:
Will `chlo::getConstantLike` generate chlo ops? it's better to use mhlo ops in such stage.

Reply:
What is the concern about?

Reply:
It will create a ConstantOp right now. But it did introduce Chlo dependences.

Reply:
following pass does not know chlo ops and these chlo ops should be legalized in the above pass pipeline. Thus, I think it's better to avoid introduce chlo ops again here.

Reply:
done